airflow/models/log.py

```diff
@@ -40,11 +40,13 @@ class Log(Base):
     owner = Column(String(500))
     owner_display_name = Column(String(500))
     extra = Column(Text)
+    try_number = Column(Integer)
 
     __table_args__ = (
         Index("idx_log_dag", dag_id),
         Index("idx_log_dttm", dttm),
         Index("idx_log_event", event),
+        Index("idx_log_task_instance", dag_id, task_id, run_id, map_index, try_number),
```

Comment:
Sorry I’ve lost context on this… did we make try_number a part of the ti’s pk? (And do we want to?) It does not look like to be the case from the current model definition, but `try_number` _is_ a part of TaskInstance’s `key`. And here we’re implying (at least from the name) that try_number is a part of the ti’s identifier.

Reply:
i don't really see it as implying that, but if you want to use a different name I don't mind.  

the PK of the TI table is what it is, and there's nothing changing that here.  

But the fact remains that the purpose here is to be able to log events specific to a try, and to be able to expose them on the UI.

But now I'm thinking we should probably just add a task_instance_event_log and put all TI-related events there, and get them out of the main log table.  this would be more efficient.  log table is already quite large.  and we could separate the events that have nothing to do with TIs from the ones that are specific to TIs.  WDYT?